api/desecapi/models/tokens.py

```diff
@@ -89,61 +89,24 @@ def generate_key(self):
     def make_hash(plain):
         return make_password(plain, salt="static", hasher="pbkdf2_sha256_iter1")
 
-    def get_policy(self, *, domain=None):
-        order_by = F("domain").asc(
-            nulls_last=True
-        )  # default Postgres sorting, but: explicit is better than implicit
+    def get_policy(self, *, domain=None, subname=None, type=None):
+        order_by = [
+            F(field).asc(
+                nulls_last=True  # default Postgres sorting, but: explicit is better than implicit
+            )
+            for field in ["domain", "subname", "type"]
+        ]
         return (
-            self.tokendomainpolicy_set.filter(Q(domain=domain) | Q(domain__isnull=True))
-            .order_by(order_by)
+            self.tokendomainpolicy_set.filter(
+                Q(domain=domain) | Q(domain__isnull=True),
+                Q(subname=subname) | Q(subname__isnull=True),
+                Q(type=domain) | Q(type__isnull=True),
```

Comment:
change to `type=type`, make sure this is tested

Reply:
done